extension/src/test/suite/AutoCompletion.test.ts

```diff
@@ -0,0 +1,273 @@
+import * as chai from "chai";
+import * as vscode from "vscode";
+import * as fs from "fs";
+import * as path from "path";
+import * as os from 'os';
+import { getLispAndDclCompletions,getCmdAndVarsCompletionCandidates } from "../../completion/autocompletionProvider";
+import { before } from "mocha";
+import { ReadonlyDocument } from "../../project/readOnlyDocument";
+import { allCmdsAndSysvars } from "../../resources";
+let assert = chai.assert;
+const testDir = path.join(__dirname + "/../../../extension/src/test");
+const outputDir = path.join(testDir + "/OutputFile");
+let LispFile = "";
+let DclFile = "";
+let lispdoc: vscode.TextDocument;
+let dcldoc: vscode.TextDocument;
+
+fs.mkdir(outputDir, { recursive: true }, (err) => {
+  if (err) {
+    return console.error(err);
+  }
+});
+
+function createFakeTextDcoument() {
+  LispFile = path.join(testDir + "/OutputFile/test.lsp");
+  DclFile = path.join(testDir + "/OutputFile/test.dcl");
+  try {
+    if (!fs.existsSync(LispFile)) {
+      fs.writeFileSync(LispFile, "");
+    }
+    if (!fs.existsSync(DclFile)) {
+      fs.writeFileSync(DclFile, "");
+    }
+    lispdoc  = ReadonlyDocument.open(LispFile);
+    dcldoc = ReadonlyDocument.open(DclFile);
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+function getSuggestLabel(doc : vscode.TextDocument ,inputword : string,isupper :boolean){
+  const suggestionList: Array<vscode.CompletionItem> = getLispAndDclCompletions(
+    doc,
+    inputword,
+    isupper
+  );
+  let suggestLabel = [];
+  suggestionList.forEach((item) => {
+    suggestLabel.push(item.label);
+  });
+  return suggestLabel;
+}
+
+function getSuggestLabelCMD(cmd :string[] ,inputword : string,isupper :boolean){
+  let suggestionList = getCmdAndVarsCompletionCandidates(cmd,inputword,isupper)
+  let suggestLabel = [];
+  suggestionList.forEach((item) => {
+    suggestLabel.push(item.label);
+  });
+  return suggestLabel;
+}
+
+suite("AutoCompletion Tests", function () {
+  // Windows only functions (vla-,vlax-,vlr-,vl-load-com,vl-load-reactors,vlisp-)
+  before(() => {
```

Comment:
it's "setup" for suite, and "before" for "describe"

Reply:
https://mochajs.org/#configuring-mocha-nodejs
![image](https://user-images.githubusercontent.com/57521938/114486498-99605780-9c40-11eb-8f80-240f483e3923.png)
![image](https://user-images.githubusercontent.com/57521938/114486539-ac732780-9c40-11eb-97d6-18a1b79acaab.png)

Reply:
Ok, I will update.